android/java/org/chromium/chrome/browser/notifications/BraveSetDefaultBrowserNotificationService.java

```diff
@@ -0,0 +1,214 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.notifications;
+
+import android.app.AlarmManager;
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.SystemClock;
+import android.provider.Settings;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.app.NotificationManagerCompat;
+
+import org.chromium.base.ContextUtils;
+import org.chromium.base.Log;
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.BraveActivity;
+
+import java.util.Calendar;
+
+public class BraveSetDefaultBrowserNotificationService extends BroadcastReceiver {
+    public Context mContext;
+    private Intent mIntent;
+
+    public static final String HAS_ASKED_AT_11_22 = "brave_set_default_browser_has_at_11_22";
+
+    public static final int NOTIFICATION_ID = 10;
+
+    // Deep links
+    public static final String DEEP_LINK = "deep_link";
+    public static final String SHOW_DEFAULT_APP_SETTINGS = "SHOW_DEFAULT_APP_SETTINGS";
+
+    // Intent types
+    public static final String INTENT_TYPE = "intent_type";
+    public static final String AT_11_22 = "at_11_22";
+
+    public static final String CANCEL_NOTIFICATION = "cancel_notification";
+
+    // Startup notification data
+    public static final String NOTIFICATION_ID_EXTRA = "notification_id_extra";
+
+    public static boolean isBraveSetAsDefaultBrowser(Context context) {
+        Intent browserIntent = new Intent("android.intent.action.VIEW", Uri.parse("http://"));
+        ResolveInfo resolveInfo = context.getPackageManager().resolveActivity(
+                browserIntent, supportsDefault() ? PackageManager.MATCH_DEFAULT_ONLY : 0);
+        return resolveInfo.activityInfo.packageName.equals(
+                       BraveActivity.BRAVE_PRODUCTION_PACKAGE_NAME)
+                || resolveInfo.activityInfo.packageName.equals(
+                        BraveActivity.BRAVE_DEVELOPMENT_PACKAGE_NAME);
+    }
+
+    private boolean shouldShowNotification() {
+        if (isBraveSetAsDefaultBrowser(mContext) || mIntent.getStringExtra(INTENT_TYPE) == null) {
+            return false;
+        }
+
+        if (mIntent.getStringExtra(INTENT_TYPE).equals(AT_11_22)) {
```

Comment:
I only port 11:22 notification in this class from https://github.com/brave/browser-android-tabs/blob/87bc3cd83a29e8b802a8f24734bf780ebe4f674a/chrome/android/java/src/org/chromium/chrome/browser/notifications/BraveSetDefaultBrowserNotificationService.java because other `INTENT_TYPE` (15min, 5, 10, 15days, and BROWSER_UPDATED) look like not used anymore in our production as far as I can tell.

Reply:
yep, correct!